### ISSUE TITLE
testsuite failure on hppa

### ISSUE THREAD

**Issue by LocutusOfBorg** [borgbackup/borg#1314]
Hi, following the error:
(note: hppa is not a release architecture, so you can just close this issue)
https://buildd.debian.org/status/fetch.php?pkg=borgbackup&arch=hppa&ver=1.0.5-2&stamp=1468279857
```
I: pybuild base:184: cd /<<PKGBUILDDIR>>/.pybuild/pythonX.Y_3.5/build; python3.5 -m pytest -k 'not benchmark and not fuse_mount' --pyargs borg.testsuite
============================= test session starts ==============================
platform linux -- Python 3.5.2, pytest-2.9.2, py-1.4.31, pluggy-0.3.1
rootdir: /<<PKGBUILDDIR>>, inifile: setup.cfg
plugins: cov-2.2.1
collected 0 items / 18 errors

==================================== ERRORS ====================================
___ ERROR collecting .pybuild/pythonX.Y_3.5/build/borg/testsuite/__init__.py ___
borg/testsuite/__init__.py:10: in <module>
    from ..xattr import get_all
borg/xattr.py:71: in <module>
    logger.error(msg)
borg/logger.py:173: in error
    return self.__logger.error(*args, **kw)
borg/logger.py:150: in __logger
    raise Exception("tried to call a logger before setup_logging() was called")
E   Exception: tried to call a logger before setup_logging() was called
___ ERROR collecting .pybuild/pythonX.Y_3.5/build/borg/testsuite/archive.py ____
/usr/lib/python3/dist-packages/py/_path/local.py:650: in pyimport
    __import__(modname)
borg/testsuite/__init__.py:10: in <module>
    from ..xattr import get_all
borg/xattr.py:71: in <module>
    logger.error(msg)
borg/logger.py:173: in error
    return self.__logger.error(*args, **kw)
borg/logger.py:150: in __logger
    raise Exception("tried to call a logger before setup_logging() was called")
E   Exception: tried to call a logger before setup_logging() was called
___ ERROR collecting .pybuild/pythonX.Y_3.5/build/borg/testsuite/archiver.py ___
/usr/lib/python3/dist-packages/py/_path/local.py:650: in pyimport
    __import__(modname)
borg/testsuite/__init__.py:10: in <module>
    from ..xattr import get_all
borg/xattr.py:71: in <module>
    logger.error(msg)
borg/logger.py:173: in error
    return self.__logger.error(*args, **kw)
borg/logger.py:150: in __logger
    raise Exception("tried to call a logger before setup_logging() was called")
E   Exception: tried to call a logger before setup_logging() was called
__ ERROR collecting .pybuild/pythonX.Y_3.5/build/borg/testsuite/benchmark.py ___
/usr/lib/python3/dist-packages/py/_path/local.py:650: in pyimport
    __import__(modname)
borg/testsuite/__init__.py:10: in <module>
    from ..xattr import get_all
borg/xattr.py:71: in <module>
    logger.error(msg)
borg/logger.py:173: in error
    return self.__logger.error(*args, **kw)
borg/logger.py:150: in __logger
    raise Exception("tried to call a logger before setup_logging() was called")
E   Exception: tried to call a logger before setup_logging() was called
___ ERROR collecting .pybuild/pythonX.Y_3.5/build/borg/testsuite/chunker.py ____
/usr/lib/python3/dist-packages/py/_path/local.py:650: in pyimport
    __import__(modname)
borg/testsuite/__init__.py:10: in <module>
    from ..xattr import get_all
borg/xattr.py:71: in <module>
    logger.error(msg)
borg/logger.py:173: in error
    return self.__logger.error(*args, **kw)
borg/logger.py:150: in __logger
    raise Exception("tried to call a logger before setup_logging() was called")
E   Exception: tried to call a logger before setup_logging() was called
___ ERROR collecting .pybuild/pythonX.Y_3.5/build/borg/testsuite/compress.py ___
/usr/lib/python3/dist-packages/py/_path/local.py:650: in pyimport
    __import__(modname)
borg/testsuite/__init__.py:10: in <module>
    from ..xattr import get_all
borg/xattr.py:71: in <module>
    logger.error(msg)
borg/logger.py:173: in error
    return self.__logger.error(*args, **kw)
borg/logger.py:150: in __logger
    raise Exception("tried to call a logger before setup_logging() was called")
E   Exception: tried to call a logger before setup_logging() was called
____ ERROR collecting .pybuild/pythonX.Y_3.5/build/borg/testsuite/crypto.py ____
/usr/lib/python3/dist-packages/py/_path/local.py:650: in pyimport
    __import__(modname)
borg/testsuite/__init__.py:10: in <module>
    from ..xattr import get_all
borg/xattr.py:71: in <module>
    logger.error(msg)
borg/logger.py:173: in error
    return self.__logger.error(*args, **kw)
borg/logger.py:150: in __logger
    raise Exception("tried to call a logger before setup_logging() was called")
E   Exception: tried to call a logger before setup_logging() was called
__ ERROR collecting .pybuild/pythonX.Y_3.5/build/borg/testsuite/hashindex.py ___
/usr/lib/python3/dist-packages/py/_path/local.py:650: in pyimport
    __import__(modname)
borg/testsuite/__init__.py:10: in <module>
    from ..xattr import get_all
borg/xattr.py:71: in <module>
    logger.error(msg)
borg/logger.py:173: in error
    return self.__logger.error(*args, **kw)
borg/logger.py:150: in __logger
    raise Exception("tried to call a logger before setup_logging() was called")
E   Exception: tried to call a logger before setup_logging() was called
___ ERROR collecting .pybuild/pythonX.Y_3.5/build/borg/testsuite/helpers.py ____
/usr/lib/python3/dist-packages/py/_path/local.py:650: in pyimport
    __import__(modname)
borg/testsuite/__init__.py:10: in <module>
    from ..xattr import get_all
borg/xattr.py:71: in <module>
    logger.error(msg)
borg/logger.py:173: in error
    return self.__logger.error(*args, **kw)
borg/logger.py:150: in __logger
    raise Exception("tried to call a logger before setup_logging() was called")
E   Exception: tried to call a logger before setup_logging() was called
_____ ERROR collecting .pybuild/pythonX.Y_3.5/build/borg/testsuite/key.py ______
/usr/lib/python3/dist-packages/py/_path/local.py:650: in pyimport
    __import__(modname)
borg/testsuite/__init__.py:10: in <module>
    from ..xattr import get_all
borg/xattr.py:71: in <module>
    logger.error(msg)
borg/logger.py:173: in error
    return self.__logger.error(*args, **kw)
borg/logger.py:150: in __logger
    raise Exception("tried to call a logger before setup_logging() was called")
E   Exception: tried to call a logger before setup_logging() was called
___ ERROR collecting .pybuild/pythonX.Y_3.5/build/borg/testsuite/locking.py ____
/usr/lib/python3/dist-packages/py/_path/local.py:650: in pyimport
    __import__(modname)
borg/testsuite/__init__.py:10: in <module>
    from ..xattr import get_all
borg/xattr.py:71: in <module>
    logger.error(msg)
borg/logger.py:173: in error
    return self.__logger.error(*args, **kw)
borg/logger.py:150: in __logger
    raise Exception("tried to call a logger before setup_logging() was called")
E   Exception: tried to call a logger before setup_logging() was called
____ ERROR collecting .pybuild/pythonX.Y_3.5/build/borg/testsuite/logger.py ____
/usr/lib/python3/dist-packages/py/_path/local.py:650: in pyimport
    __import__(modname)
borg/testsuite/__init__.py:10: in <module>
    from ..xattr import get_all
borg/xattr.py:71: in <module>
    logger.error(msg)
borg/logger.py:173: in error
    return self.__logger.error(*args, **kw)
borg/logger.py:150: in __logger
    raise Exception("tried to call a logger before setup_logging() was called")
E   Exception: tried to call a logger before setup_logging() was called
___ ERROR collecting .pybuild/pythonX.Y_3.5/build/borg/testsuite/lrucache.py ___
/usr/lib/python3/dist-packages/py/_path/local.py:650: in pyimport
    __import__(modname)
borg/testsuite/__init__.py:10: in <module>
    from ..xattr import get_all
borg/xattr.py:71: in <module>
    logger.error(msg)
borg/logger.py:173: in error
    return self.__logger.error(*args, **kw)
borg/logger.py:150: in __logger
    raise Exception("tried to call a logger before setup_logging() was called")
E   Exception: tried to call a logger before setup_logging() was called
___ ERROR collecting .pybuild/pythonX.Y_3.5/build/borg/testsuite/platform.py ___
/usr/lib/python3/dist-packages/py/_path/local.py:650: in pyimport
    __import__(modname)
borg/testsuite/__init__.py:10: in <module>
    from ..xattr import get_all
borg/xattr.py:71: in <module>
    logger.error(msg)
borg/logger.py:173: in error
    return self.__logger.error(*args, **kw)
borg/logger.py:150: in __logger
    raise Exception("tried to call a logger before setup_logging() was called")
E   Exception: tried to call a logger before setup_logging() was called
__ ERROR collecting .pybuild/pythonX.Y_3.5/build/borg/testsuite/repository.py __
/usr/lib/python3/dist-packages/py/_path/local.py:650: in pyimport
    __import__(modname)
borg/testsuite/__init__.py:10: in <module>
    from ..xattr import get_all
borg/xattr.py:71: in <module>
    logger.error(msg)
borg/logger.py:173: in error
    return self.__logger.error(*args, **kw)
borg/logger.py:150: in __logger
    raise Exception("tried to call a logger before setup_logging() was called")
E   Exception: tried to call a logger before setup_logging() was called
_ ERROR collecting .pybuild/pythonX.Y_3.5/build/borg/testsuite/shellpattern.py _
/usr/lib/python3/dist-packages/py/_path/local.py:650: in pyimport
    __import__(modname)
borg/testsuite/__init__.py:10: in <module>
    from ..xattr import get_all
borg/xattr.py:71: in <module>
    logger.error(msg)
borg/logger.py:173: in error
    return self.__logger.error(*args, **kw)
borg/logger.py:150: in __logger
    raise Exception("tried to call a logger before setup_logging() was called")
E   Exception: tried to call a logger before setup_logging() was called
___ ERROR collecting .pybuild/pythonX.Y_3.5/build/borg/testsuite/upgrader.py ___
/usr/lib/python3/dist-packages/py/_path/local.py:650: in pyimport
    __import__(modname)
borg/testsuite/__init__.py:10: in <module>
    from ..xattr import get_all
borg/xattr.py:71: in <module>
    logger.error(msg)
borg/logger.py:173: in error
    return self.__logger.error(*args, **kw)
borg/logger.py:150: in __logger
    raise Exception("tried to call a logger before setup_logging() was called")
E   Exception: tried to call a logger before setup_logging() was called
____ ERROR collecting .pybuild/pythonX.Y_3.5/build/borg/testsuite/xattr.py _____
/usr/lib/python3/dist-packages/py/_path/local.py:650: in pyimport
    __import__(modname)
borg/testsuite/__init__.py:10: in <module>
    from ..xattr import get_all
borg/xattr.py:71: in <module>
    logger.error(msg)
borg/logger.py:173: in error
    return self.__logger.error(*args, **kw)
borg/logger.py:150: in __logger
    raise Exception("tried to call a logger before setup_logging() was called")
E   Exception: tried to call a logger before setup_logging() was called
========================== 18 error in 35.34 seconds ===========================
E: pybuild pybuild:274: test: plugin distutils failed with: exit code=1: cd /<<PKGBUILDDIR>>/.pybuild/pythonX.Y_3.5/build; python3.5 -m pytest -k 'not benchmark and not fuse_mount' --pyargs borg.testsuite
dh_auto_test: pybuild --test -i python{version} -p 3.5 returned exit code 13
debian/rules:63: recipe for target 'override_dh_auto_test' failed
make[1]: *** [override_dh_auto_test] Error 25
make[1]: Leaving directory '/<<PKGBUILDDIR>>'
debian/rules:46: recipe for target 'build-arch' failed
make: *** [build-arch] Error 2
dpkg-buildpackage: error: debian/rules build-arch gave error exit status 2
```

**Comment by enkore** [comment]
I'm not sure why it doesn't find the libc, but I'll at least fix the error handling there.

According to the file list: https://packages.debian.org/sid/hppa/libc6/filelist it's just libc.so.6 like everywhere else... hm...

**Comment by LocutusOfBorg** [comment]
sorry I can't test, because non-official Debian release architectures have no access as porterbox for developers.

**Comment by enkore** [comment]
Ok. How does a "round trip" look like for code to get there? Only released versions?

**Comment by LocutusOfBorg** [comment]
I have to upload a new borgbackup release on unstable to test it, not sure if worth building it everywhere and make people update it.
But if you want to include into an .6 release... :) we will know the result
(or ask me to upload it right now, and I'll comply!)

**Comment by enkore** [comment]
Ok, even though it's "unstable" unreleased version should probably not go there :)

**Comment by LocutusOfBorg** [comment]
we have also experimental for such tests... I'll upload it there, adding the two patches below:
https://github.com/borgbackup/borg/commit/98530fffd3f81977a03c838167921d0d58f697c4.patch

https://github.com/borgbackup/borg/commit/199285b2ba74a9ab789cefb8ec3e2fde1f269a40.patch

**Comment by LocutusOfBorg** [comment]
new failure:
https://buildd.debian.org/status/fetch.php?pkg=borgbackup&arch=hppa&ver=1.0.5-3&stamp=1468323220

**Comment by ThomasWaldmann** [comment]
```
E   Exception: Can't find C library [libfakeroot-sysv.so:libeatmydata.so: cannot open shared object file: No such file or directory]. Try installing ldconfig, gcc/cc or objdump.
```
I've seen the code that tries to locate libraries - by using one of the mentioned tools (wondering whether there is no better method). So, if not at least one of them is installed, it'll blow up like seen there.

**Comment by enkore** [comment]
IIRC there was a fix in master to allow eatmydata to work with Borg.

**Comment by enkore** [comment]
#1188

**Comment by enkore** [comment]
For backport of #1188 see #1320, which should fix this issue in the 1.0.x series.
